### PR TITLE
Simplified component structure, widget error boundary, removed support for React v15

### PR DESCRIPTION
**Pull request by wafflepie**
React v15 cannot be supported because we use the `createPortal` function. 

I removed the `Widgets` component, because it didn't really make sense (the check fails only if there is a mistake in our code, hence we shouldn't rely on it). `WidgetPortal` component was renamed to `Widget`, because the portal functionality is pretty simple and the name `WidgetPortal` is not as intuitive when you come to think: "What's the entry point for a widget?".

I also wrapped the `Widget` component in an error boundary HOC, because if an error occurs in a single widget, it doesn't unmount the whole application.

### CODE REVIEW

File: packages/react-union/src/components/Union/Union.js [hunk 1]
```diff
@@ -74,15 +74,14 @@ class Union extends Component {
...
+	);
 
 	render() {
-		return (
-			<div>
-				{this.props.children}
-				<Widgets configs={this.state.configs} />
-			</div>
-		);
+		return [this.props.children, this.state.configs.map(this.renderWidget)];
```
Comment (tommmyy) [inline packages/react-union/src/components/Union/Union.js]: Return one div element not an array. I wanna Union to be compatible with component providers (e.g.  react-redux's Provider component or IntlProvider, ...) that are using `Children.only` validation.

File: packages/react-union/src/decorators/withErrorBoundary/withErrorBoundary.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+		state = {
+			hasError: false,
+		};
+
+		componentDidCatch() {
+			this.setState({ hasError: true });
+		}
+
+		render() {
+			const { name } = this.props.descriptor;
```
Comment (tommmyy) [inline packages/react-union/src/decorators/withErrorBoundary/withErrorBoundary.js]: use 'path' from Ramda to get `name`.

File: packages/react-union/src/components/Union/Union.js (hunk 1 above)
Comment (wafflepie) [reply packages/react-union/src/components/Union/Union.js]: Good call. At first, I wanted to use React.Fragment but I didn't realise it's supported from React v16.2, so I replaced it with an array.

File: packages/react-union/src/decorators/withErrorBoundary/withErrorBoundary.js [hunk 3]
```diff
@@ -15,7 +16,7 @@ const withErrorBoundary = WrappedComponent => {
 		}
 
 		render() {
-			const { name } = this.props.descriptor;
+			const name = path(['props', 'descriptor', 'name'], this);
```
Comment (tommmyy) [inline packages/react-union/src/decorators/withErrorBoundary/withErrorBoundary.js]: refactor to `const getName = path(['props', 'descriptor', 'name'])` and move it above the component. Here just call `getName(this)`

Comment (wafflepie) [reply packages/react-union/src/decorators/withErrorBoundary/withErrorBoundary.js]: Yeah, that's better. Thanks!

Comment (tommmyy) [reply packages/react-union/src/decorators/withErrorBoundary/withErrorBoundary.js]: just to make it bit quicker

**Pull request merged**